app/views/ProjectsView/components/AddEditProjectDialog/index.tsx

```diff
@@ -56,33 +57,31 @@ const AddEditProjectDialog = ({
   const [users, setUsers] = useState<UserType[]>([]);
   const [reports, setReports] = useState<ShortReportType[]>([]);
   const [existingReports, setExistingReports] = useState<ShortReportType[]>([]);
-
-  const [isReportsLoading, setIsReportsLoading] = useState(true);
-  const [isSaving, setIsSaving] = useState(false);
   const { adminAccess, managerAccess } = useResource();
-  useEffect(() => {
-    if (isOpen) {
-      const getData = async () => {
-        try {
-          const resp = await api.get(`/project/${editData.ident}/reports`).request();
-          // Existing reports is solely for calculating difference for submit API call
-          setExistingReports(resp);
-          setReports(resp);
-        } catch (err) {
-          console.error(err);
-          snackbar.error('Error: failed to get reports');
-        } finally {
-          setIsReportsLoading(false);
-        }
-      };
+  const queryClient = useQueryClient();
 
+  const { isLoading: isReportsLoading, refetch } = useQuery<ShortReportType[]>({
+    queryKey: ['projectReports', editData?.ident],
+    queryFn: async () => {
       if (editData?.ident) {
-        getData();
-      } else {
-        setIsReportsLoading(false);
+        const resp = await api.get(`/project/${editData.ident}/reports`).request();
+        return resp;
       }
+      return [];
+    },
+    enabled: Boolean(isOpen && editData?.ident),
+    onSuccess: (data) => {
+      setExistingReports(data);
+      setReports(data);
+    },
+    onError: () => snackbar.error('Error: failed to get reports'),
+  });
+
+  useEffect(() => {
```

Comment:
I'm not too sure why this is needed here, wouldn't `useQuery` take care of this?